CONTRIBUTING.md

```diff
@@ -1,30 +1,112 @@
-# Contributing Guidelines
+# Contributing Guidelines and Workflow
 
-Thank you for your interest in contributing to our project. Whether it's a bug report, new feature, correction, or additional
-documentation, we greatly value feedback and contributions from our community.
+Thank you for your interest in contributing to our project. Whether it's a bug report, new feature, correction, or additional documentation, we greatly value feedback and contributions from our community.
 
-Please read through this document before submitting any issues or pull requests to ensure we have all the necessary
-information to effectively respond to your bug report or contribution.
+Please read through this document before editing the code base, submitting any issues, or submitting pull requests to ensure we have all the necessary information to effectively respond to your bug report or contribution.
 
+**In this document:**
 
-## Reporting Bugs/Feature Requests
+- [Contributor Guidelines](#contributor-guidelines)
+- [Contributor Workflow](#contributor-workflow)
+- [Maintainer Workflow](#maintainer-workflow)
+
+---
+
+## [Contributor Guidelines](#contributor-guidelines)
+
+### Reporting Bugs or Feature Requests
 
 We welcome you to use the GitHub issue tracker to report bugs or suggest features.
 
-When filing an issue, please check existing open, or recently closed, issues to make sure somebody else hasn't already
-reported the issue. Please try to include as much information as you can. Details like these are incredibly useful:
+When filing an issue, please check existing open, or recently closed, issues to make sure somebody else hasn't already reported the issue. Please try to include as much information as you can. Details like these are incredibly useful:
 
 * A reproducible test case or series of steps
 * The version of our code being used
 * Any modifications you've made relevant to the bug
 * Anything unusual about your environment or deployment
 
+### Reporting Security Issues
+
+If you discover a potential security issue in this project we ask that you notify AWS/Amazon Security via our [vulnerability reporting page](http://aws.amazon.com/security/vulnerability-reporting/). Please do **not** create a public GitHub issue.
+
+
+### Finding Contributions to Work On
```

Comment:
I just confirmed with a few style guides. The AP, NYT, and Chicago style guides all say "On" should be capitalized because it is the last word in the title. I'll leave it as-is.